File: regexlib/javascript_test_file/regexlib_5511.js
/* 5511
 * ^([a-z]+?\.[a-z]+)+\%$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"aaa."*32+"!1 __EOA(i or ii)"
 */
var REGEX = /^([a-z]+?\.[a-z]+)+\%$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'aaa.'.repeat(i*1) + '!1 __EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_5511.py
# 5511
# ^([a-z]+?\.[a-z]+)+\%$
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"aaa."*32+"!1 __EOA(i or ii)"

import re
from time import perf_counter

regex = """^([a-z]+?\.[a-z]+)+\%$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "aaa." * i * 1 + "!1 __EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")